omnigibson/utils/vision_utils.py

```diff
@@ -106,3 +106,65 @@ def segmentation_to_rgb(seg_im, N, colors=None):
         return (255.0 * use_colors[seg_im]).astype(np.uint8)
     else:
         return (use_colors[seg_im]).astype(np.float)
+
+class Remapper:
+    def __init__(self):
+        self.key_array = np.array([], dtype=np.uint32)  # Initialize the key_array as empty
+        self.known_ids = set()
+
+    def clear(self):
+        """Resets the key_array to empty."""
+        self.key_array = np.array([], dtype=np.uint32)
+        self.known_ids = set()
+
+    def remap(self, old_mapping, new_mapping, image):
+        """
+        Remaps values in the given image from old_mapping to new_mapping using an efficient key_array.
+        
+        Args:
+            old_mapping (dict): The old mapping dictionary that maps a set of image values to labels, e.g. {1:'desk',2:'chair'}.
+            new_mapping (dict): The new mapping dictionary that maps another set of image values to labels, e.g. {5:'desk',7:'chair',9:'sofa'}.
```

Comment:
nit, this is not entirely accurate, right? old_mapping and new_mapping should have the same number of keys?
e.g. if the old mapping is {1:'desk',2:'chair'}, new mapping can be {5:'desk',7:'chair'}. The sofa shouldn't show up here?